test/integration/targets/mongodb_replicaset/tasks/main.yml

```diff
@@ -0,0 +1,438 @@
+# test code for the mongodb_replicaset module
+# (c) 2019,  Rhys Campbell <rhys.james.campbell@googlemail.com>
+
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+# ============================================================
+
+- name: Ensure tests home exists
+  file:
+    path: /home/tests
+    state: directory
+
+- name: Execute cleanup script if /home/tests/mongodb[1-3] directories exist
+  script: files/bash/mongod_teardown.sh
```

Comment:
Instead of invoking shell scripts, include a file with the necessary tasks.

Reply:
Bash scripts have been replaced with a bunch of include_tasks and other bits. Not really on board with the point of this as I think it's one of the rare occasions where bash scripts are a more elegant solution than Ansible tasks. 

Can you have a look and comment if I'm on the right track. I've made these changes for the mongodb_replicaset tests only. mongodb_shard tests will be a little more complicated since there's multiple replicasets, config server and so on.

Reply:
Yes, that's much better.

Using include tasks will make it easier to use `remote_tmp_dir` instead of `/home/tests` as well.

Reply:
All /home/tests paths repointed to use remote_tmp_dir.